program/src/instructions/compressed/close_expired_listing.rs

```diff
@@ -50,6 +52,7 @@ pub fn process_close_expired_listing<'info>(
         TcompError::ListingNotYetExpired
     );
 
+    // TODO: asset_id check. Shoujld compare asset_id that you get with these params with the one actually stored in listing.
```

Comment:
User provides a list state address and merkle tree and nonce. We derive the asset id from that and derive the correct list state and check it matches what the user passes in. As long as these parameters match and the list state is expired, it can be closed.